lib/Doctrine/ORM/EntityManager.php

```diff
@@ -784,7 +773,7 @@ public function newHydrator($hydrationMode)
                 return new Internal\Hydration\SimpleObjectHydrator($this);
 
             default:
-                $class = $this->config->getCustomHydrationMode($hydrationMode);
+                $class = $this->config->getCustomHydrationMode((string) $hydrationMode);
```

Comment:
This is most likely wrong, see https://github.com/doctrine/doctrine2/pull/7130#discussion_r175259739. It could be `int|string` currently.

Reply:
getCustomHydrationMode expect a string in it's function definition.

Reply:
:thinking:

Reply:
Changed this in my PR since it's really `int|string` now. Something to re-do later on probably.